src/core_plugins/kibana/server/tutorials/apm/saved_objects.js

```diff
@@ -0,0 +1,309 @@
tring\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"sourcemap.service.name\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"sourcemap.service.version\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.duration.us\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.id\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.name\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.parent\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.start.us\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"span.type\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"tags\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.duration.us\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.id\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.name\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":false,\"readFromDocValues\":false},{\"name\":\"transaction.name.keyword\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.result\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.sampled\",\"type\":\"boolean\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.span_count.dropped.total\",\"type\":\"number\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"transaction.type\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"view errors\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true},{\"name\":\"view spans\",\"type\":\"string\",\"count\":0,\"scripted\":false,\"searchable\":true,\"aggregatable\":true,\"readFromDocValues\":true}]"
```

Comment:
Is this just for debugging purposes? I assume this should be dynamically generated, so that when the mapping changes this doesn't get out of sync.

Reply:
Storing the index pattern saved object like this is on purpose. If the mappings change between versions then this will need to be updated - just like the other saved objects will need to be updated in the event that field names change and so on.

Reply:
True, we still need to update the dashboards if a field is renamed or removed. But up until now we've been able to make backwards compatible changes to the schema (like adding a new field) that didn't require us to update anything in Kibana.

What is the advantage of storing the static object, instead of creating it dynamically?

Reply:
The advantage is being able to link the other saved objects directly to the index-pattern by id. How does the import work now, how do the saved objects link to the index-pattern? Are the users prompted with a list of index-patterns that match the index-pattern title?

Reply:
The user might access APM UI (which also needs the index pattern) before opening the Getting Started Guide. We need a single solution for creating the index patterns, so we don't end up with duplicate saved objects files that needs to be updated whenever apm-server schema changes.

I outlined three different solutions here: https://github.com/elastic/apm-server/issues/991

Reply:
After a lengthy discussion we agreed to leave the dashboards and index patterns in Kibana (and not import it via APM-Server). 

That's good news since this can largely stay the same.

A few suggestions: 
 - Can the dashboards and index patterns live in separate json files? This will make it possible for us to update them automatically via a build step in the future.
 - In addition to the "import dashboards"-button, we also need an "import index patterns"-button. People might not want the dashboards, but will need index patterns for APM to work correctly.

Reply:
> Can the dashboards and index patterns live in separate json files? This will make it possible for us to update them automatically via a build step in the future.

Yes

> In addition to the "import dashboards"-button, we also need an "import index patterns"-button. People might not want the dashboards, but will need index patterns for APM to work correctly.

Do we really want to expose that level of implementation details to the user? What is the harm in importing everything with a single action? Seems like the harm is less than the complexity of having two buttons that require a deeper knowledge about how kibana works to understand why you would click one or the other or both.
It might be a benefit to install all saved objects. That way the other kibana apps will never be empty.

Reply:
@nreese You might be right. The thinking behind having two buttons ["Import index pattern", "Import index pattern and dashboards"] is that users will need to manually update the index pattern whenever they update their version of the stack. However, clicking "Import index pattern and dashboards" will also overwrite existing visualizations (AFAIK, correct me if I'm wrong), which can be annoying. 
But, we might be overthinking this. Thoughts?

Reply:
Bit more context: The reason for mentioning "Index pattern" specifically is because we have this call out to the user in the UI: https://github.com/elastic/kibana/issues/20077
Having said that, I like what you did with abstracting all the details away by writing "Import Kibana objects". Maybe that's much better. We can have the details on a documentation page, or something. @formgeist @sqren Thoughts?

Reply:
+1 for just having one option and calling it "Import Kibana objects".

Reply:
Fine with me. If this causes issues for the users we can iterate on it.